airflow/cli/commands/connection_command.py

```diff
@@ -82,25 +83,35 @@ def connections_list(args):
         )
 
 
+def _connection_to_dict(conn: Connection) -> dict:
+    return dict(
+        conn_type=conn.conn_type,
+        description=conn.description,
+        login=conn.login,
+        password=conn.password,
+        host=conn.host,
+        port=conn.port,
+        schema=conn.schema,
+        extra=conn.extra,
```

Comment:
What about dejson? Wouldn't this likely result in the extra field being a string containing a JSON dict?, i.e. `"extra": "{\"extra_key\": \"value\"}"`

(I'm not sure there's a ready solution to this, as it's _possible_ the extra isn't JSON

Reply:
it doesn't matter what is stored in `extra` because it is a string and when we serialize it we don't make any assumptions about what it contains.

when we deserialize it, it is again a string and we pass it to Connection as a kwarg.

So yes suppose we have this connection

```
Connection(extra=json.dumps({'hi': 'bye'}))
```
So the CLI exporter will export this like so
```
'{"extra": "{\\"hi\\": \\"bye\\"}"}'
```
So when we try to parse this again we do json.loads, and now we get a dict where `extra` is a plain json string, as desired:
```
{'extra': '{"hi": "bye"}'}
```
So, everything is cool right?
There's one more detail here though.
What we _do also_ support here, is serializing airflow conn extra fully as object.
By that I mean, storing _this_ uri _is permissible_:
```
'{"extra": {"hi": "bye"}}'
```
And the reason we can support this is, if it's stored like this, `extra` will be `dict` when the value is loaded.
We just can't _serialize_ in this way by default because there is no guarantee that `extra` will be json.

So in sum, we support `[obj, str] > Connection.extra`, but when _generating_ the value in CLI export, we have to only support `Connection.extra -> str`.

I _wish_ this were not the case, but because we don't require extra to be json (maybe until 3.0?) i think that's how it must be.

Reply:
> We just can't serialize in this way by default because there is no guarantee that extra will be json

Yeah, that's sort of what I was getting it.

I think we should work out a deprecation plan for storing anything but json in extra.

Reply:
> I think we should work out a deprecation plan for storing anything but json in extra.

yeah i think that would be good